pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -185,20 +185,17 @@ def _run_optimization(
         else:
             if self.trainer.lightning_module.automatic_optimization:
                 self._optimizer_step(optimizer, opt_idx, batch_idx, closure)
-                if len(self.trainer.optimizers) > 1:
```

Comment:
This assumes len() == 2, consider 3 optimizers - it should be a cyclic toggle instead of boolean